chia/farmer/farmer.py

```diff
@@ -153,6 +153,13 @@ async def setup_keys(self) -> bool:
             log.warning(no_keys_error_str)
             return False
 
+        config = load_config(self._root_path, "config.yaml")
+        self.config = config["farmer"]
```

Comment:
I think this is the reason the test is failing. `xch_target_address` is already present in `self.config` in the test, but loading it here, and replacing `self.config` removes it.

Reply:
this makes the test pass for me:

```
+        config = load_config(self._root_path, "config.yaml")
+        if "xch_target_address" not in self.config:
+            self.config = config["farmer"]
+
+        if "xch_target_address" not in self.pool_config:
+            self.pool_config = config["pool"]
+
+        if "xch_target_address" not in self.config or "xch_target_address" not in self.pool_config:
+            log.debug("xch_target_address missing in the config")
+            return False
+
```